### PR TITLE
feat(SecondaryToken): add a SecondaryNEP5Token

### PR DESCRIPTION
**Pull request by davemneo**
Overview:
Adding a SecondaryNEP5Token with the ability to assign another contract as it's `primary` caller, enabling remote minting calls from the `primary` caller.

### Description of the Change
Added SecondaryNEP5Token: A NEP5Token that extends a Secondary(SmartContract)
Moved NEP5 and the new SecondaryNEP5 token to a Token folder.
Created a MintableToken interface to be used by a calling contract thusly:
    const smartContract = SmartContract.for<MintableToken>();

### Test Plan

    yarn jest SecondaryNEP5

### CODE REVIEW

File: packages/neo-one-smart-contract-lib/src/token/SecondaryNEP5Token.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  MapStorage,
+  SmartContract,
+} from '@neo-one/smart-contract';
+
+interface TokenPayableContract {
+  readonly approveReceiveTransfer: (from: Address, amount: Fixed<0>, asset: Address) => boolean;
+  readonly onRevokeSendTransfer: (from: Address, amount: Fixed<0>, asset: Address) => void;
+}
+
+export function SecondaryNEP5Token<TBase extends Constructor<SmartContract>>(Base: TBase) {
```
Comment (dicarlo2) [inline packages/neo-one-smart-contract-lib/src/token/SecondaryNEP5Token.ts]: I must be missing something - why is this called `SecondaryNEP5Token`? How does this differ from the old `NEP5Token` mixin?

Comment (davemneo) [reply packages/neo-one-smart-contract-lib/src/token/SecondaryNEP5Token.ts]: This is a work in progress.  Sorry, should have commented in the notes.

File: packages/neo-one-smart-contract-lib/src/__data__/contracts/token/TestSecondaryNEP5Token.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    email: 'alex.dicarlo@neotracker.io',
+    description: 'The TestSecondaryNEP5Token',
+  };
+  public constructor(
+    owner: Address = Deploy.senderAddress,
+    amount: Fixed<8> = 1_000_000_00000000,
+    protected readonly initialPrimary = Deploy.senderAddress,
+  ) {
+    super();
+    this.initialPrimary = owner;
```
Comment (dicarlo2) [inline packages/neo-one-smart-contract-lib/src/__data__/contracts/token/TestSecondaryNEP5Token.ts]: Remove and just set it above: `protected readonly initialPrimary = owner`

File: packages/neo-one-smart-contract-lib/src/__data__/contracts/token/TestSecondaryNEP5Token.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    if (!Address.isCaller(owner)) {
+      throw new Error('Sender was not the owner.');
+    }
+    this.owner = owner;
```
Comment (dicarlo2) [inline packages/neo-one-smart-contract-lib/src/__data__/contracts/token/TestSecondaryNEP5Token.ts]: Remove: `public readonly owner` as a parameter

File: packages/neo-one-smart-contract-lib/src/__data__/contracts/token/TestSecondaryNEP5Token.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    this.issue(owner, amount);
+  }
+  public primaryOnlyAction() {
```
Comment (dicarlo2) [inline packages/neo-one-smart-contract-lib/src/__data__/contracts/token/TestSecondaryNEP5Token.ts]: What's the point of this method?

File: packages/neo-one-smart-contract-lib/src/__tests__/examples/RedToken.test.ts [hunk 5]
```diff
@@ -1,14 +1,14 @@
 import BigNumber from 'bignumber.js';
 import * as path from 'path';
-import { testToken } from '../__data__';
+import { tokenTester } from '../../__data__';
```
Comment (dicarlo2) [inline packages/neo-one-smart-contract-lib/src/__tests__/examples/RedToken.test.ts]: Why was this renamed from `testToken` to `tokenTester`?

File: packages/neo-one-smart-contract-lib/src/token/SecondaryNEP5Token.ts [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  SmartContract,
+} from '@neo-one/smart-contract';
+import { Secondary } from '../ownership';
+
+interface TokenPayableContract {
+  readonly approveReceiveTransfer: (from: Address, amount: Fixed<0>, asset: Address) => boolean;
+  readonly onRevokeSendTransfer: (from: Address, amount: Fixed<0>, asset: Address) => void;
+}
+
+export abstract class SecondaryNEP5Token extends Secondary(SmartContract) {
```
Comment (dicarlo2) [inline packages/neo-one-smart-contract-lib/src/token/SecondaryNEP5Token.ts]: It's not clear to me what adding `Secondary` here does. We don't appear to be using the mixin anywhere in this class.
